src/client/contextChannels.ts

```diff
@@ -2,79 +2,424 @@
  * @module ContextChannels
  */
 
+/* eslint-disable no-dupe-class-members */
+
+import {EventEmitter} from 'events';
+
 import {Identity} from 'openfin/_v2/main';
 
-import {tryServiceDispatch} from './connection';
-import {APIFromClientTopic} from './internal';
+import {tryServiceDispatch, eventEmitter} from './connection';
+import {APIFromClientTopic, ChannelTransport, DesktopChannelTransport} from './internal';
+import {Context} from './context';
+import {ContextListener} from './main';
 
 export type ChannelId = string;
 
-export const GLOBAL_CHANNEL_ID: ChannelId = 'global';
+export const DEFAULT_CHANNEL_ID: ChannelId = 'default';
 
 /**
- * Object representing a fully-initialised channel.
+ * Event fired when a window is added to a channel. See {@link Channel.addEventListener}.
+ *
+ * Note that this event will typically fire as part of a pair - since windows must always belong to a channel, a window
+ * can only join a channel by leaving it's previous channel. The exceptions to this rule are when the window is created
+ * and destroyed when there will be no previous channel or no current channel, respectively.
  *
- * Whenever the service returns any kind of channel info, it'll always be with this type.
+ * @event
  */
-export interface Channel {
+export interface ChannelWindowAddedEvent {
+    type: 'window-added';
+
     /**
-     * Would be generated by the service - what you pass to the API calls below
+     * The window that has just been added to the channel.
      */
-    id: ChannelId;
+    identity: Identity;
 
     /**
-     * Uniquely defines each channel type
+     * The channel that has been modified. Will always be the channel object that {@link Channel.addEventListener} was
```

Comment:
This is pointing out that if you have a channel "`c`", and you do `c.addEventListener('window-added', ...)`, the `channel` property of all the events you receive will always be `c` (as in "triple-equal to" `c`, not just an object with the same channel ID).

It's not saying that only `channel` has been modified, just that `channel` will be the channel you're listening to. The same is true of `window-removed`, only with `previousChannel` rather than `channel`.

Perhaps this is unnecessary, or more misleading than helpful. Wording is clunky, but that was the best I could come up with - can discuss.

Reply:
To be clear, my issue is with the first sentence, not the second